presto-base-arrow-flight/src/test/java/com/facebook/plugin/arrow/TestArrowFlightEchoQueries.java

```diff
@@ -149,21 +151,12 @@ public void close()
         allocator.close();
     }
 
-    public static int getServerPort()
-    {
-        return 9444;
-    }
-
-    public static Location getServerLocation()
-    {
-        return Location.forGrpcTls("localhost", getServerPort());
-    }
-
     @Override
     protected QueryRunner createQueryRunner()
             throws Exception
     {
-        return ArrowFlightQueryRunner.createQueryRunner(9444);
+        serverPort = ArrowFlightQueryRunner.findUnusedPort();
```

Comment:
If you need serverport elsewhere can we move it to the constructor else we don't need to pass a random port

Reply:
Yes, `serverPort` is also needed in `setup()` when starting the server. I agree, this is a little awkward because `createQueryRunner` is called from the base class init() first. I'll try assigning the port in the constructor.